lib/libutee/tee_api_objects.c

```diff
@@ -397,20 +397,43 @@ TEE_Result TEE_OpenPersistentObject(uint32_t storageID, void *objectID,
 				    uint32_t objectIDLen, uint32_t flags,
 				    TEE_ObjectHandle *object)
 {
-	if (storageID != TEE_STORAGE_PRIVATE)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	TEE_Result res;
 
-	if (objectID == NULL)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	if (storageID != TEE_STORAGE_PRIVATE) {
+		res = TEE_ERROR_ITEM_NOT_FOUND;
```

Comment:
I agree the way it is currently written is redundant.

But it clearly shows the GP API, with accepted return codes and panic case, that are not spread everywhere in the code, but located only at the end of the function.  So I am quite in favor of this approach.

We may need a referee  :)  ?